spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/StatusReportsController.java

```diff
@@ -91,6 +93,9 @@ public ModelAndView pluginStatusReport(Request request, Response response) {
         } catch (UnsupportedOperationException e) {
             String message = String.format("Status Report for plugin with id: '%s' is not found.", pluginId);
             return errorPage(response, 404, "Plugin Status Report", message);
+        } catch (RulesViolationException | SecretResolutionFailureException e) {
+            LOGGER.error(e.getMessage(), e);
+            return errorPage(response, 422, "Plugin Status Report", e.getMessage());
```

Comment:
This should not be `422`, it's not a user error.

Reply:
The exception can be raised due to a secret being configured for a non-existent secret config id. Which is a user error.
If not 422, then what? 500?

Reply:
> The exception can be raised due to a secret being configured for a non-existent secret config id. Which is a user error.

These would be user errors during config update right.

Reply:
Yes but these are not caught during full config save.